### PR TITLE
gyp: manually configure sodium.linux.gypi for arm

### PR DESCRIPTION
**Pull request by rvagg**
/R=@thlorenz

we probably should collapse this into the `nsolid` branch now rather than `thlorenz/gyp` too eh?

### CODE REVIEW

**Comment by thlorenz** [comment]
## TL;DR

This change will work in the short run, but will most likely break further down the road when we update to later libsodium versions. It may be fine though as we should consider switching to the lower level bindings by @mafintosh.

## More Details

Not sure if that is going to work in the long run :(
The gyp file is actually generated [via a script](https://github.com/nodesource/libsodium/blob/nsolid/gen-gypi.py).
The script was run on OSX and Linux.

This is to avoid people having to install automake and such.

So this will most likely work in the short run, but may break every time we upgrade to a new libsodium version.
In the long run we should update the script to generate the gypfile in the specific way (or similar) to what you updated in there.

### Options to switch to lower level bindings

I sat with @mafintosh as he created his [lower level sodium bindings](https://github.com/mafintosh/sodium-native). He actually used my script as a starting point but afaik he improved on it, however it requires node to be installed: [makefile.js](https://github.com/mafintosh/sodium-native/blob/master/makefile.js).

I believe his abstractions are better than the libsodium ones as the latter adds too much extra on top, while his are just clean bindings. So in the long run we should investigate switching to his bindings.

If we make that decision, we need to evaluate though that things actually work on all platforms.
Afaik he's running the `./configure` command inside libsodium to have all sources and the Makefile generated. Then he [includes needed sources dynamically into the libsodium.gyp file](https://github.com/mafintosh/sodium-native/blob/master/deps/libsodium.gyp#L73) and does this similarly for [cflags](https://github.com/mafintosh/sodium-native/blob/master/deps/libsodium.gyp#L59) and [defines](https://github.com/mafintosh/sodium-native/blob/master/deps/libsodium.gyp#L14).

**Comment by rvagg** [comment]
All correct, but given that a diff of the regeneration will show that there's some custom work in here and the builds will fail, it should be a good enough signal that we have another manual step to do. There's not a whole lot we can do other than run the gyp generation on build which is a mess with the dependencies.

Migration to newer bindings would be great but that's not in scope for _just making this work_!

**Comment by thlorenz** [comment]
OK @rvagg as long as the changes to the gyp file don't break any other platforms, mainly Linux and OSX  - proof read the changes and looks like it doesn't break anything - than I'm ok to merge for now, but we should really improve on the auto generation in the future.

So just merge to master and talk to @brycebaril about tagging this and updating N|Solid to pick it up.
